Src/HALALMock/Services/SharedMemory/SharedMemory.cpp

```diff
@@ -1,9 +1,60 @@
 
 #include "HALALMock/Services/SharedMemory/SharedMemory.hpp"
+//includes to create the shared Memory in Posix
+#include <fcntl.h>
+#include <sys/shm.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <unistd.h>
 
 void SharedMemory::start() {
     // TODO: create / open shared memory
 }
+
+std::string SharedMemory::generate_shared_memory_name(){
+	return "SharedMemory_emulated_state_machine_"+std::to_string(getpid());
```

Comment:
this wont work if another process wants to connect to it, because it wont know the PID